### ISSUE TITLE
Warnings about hooks usage with React 18

### ISSUE THREAD

**Issue by itmanyong** [microsoft/griffel#89]
React's hook limit was triggered!!   cli：vitev2.9+reactv18
![image](https://user-images.githubusercontent.com/53500371/161308077-dbbb5ad0-16a3-47ba-ab2a-0f1ae0f03a74.png)

**Comment by layershifter** [comment]
I checked it and it works, https://stackblitz.com/edit/vitejs-vite-3pwswk?file=src%2FApp.jsx&terminal=dev

---

Please provider more details, reproducible demo will help a lot.

**Comment by itmanyong** [comment]
> 我检查了它，它的工作原理，https://stackblitz.com/edit/vitejs-vite-3pwswk?file=src%2FApp.jsx&terminal=dev
> 
> **编辑：**也适用于 React 18，https://stackblitz.com/edit/vitejs-vite-rvdrxe?file=src%2FApp.jsx&terminal=dev
> 
> 请提供更多详细信息，可重复的演示将有很大帮助。
Maybe I didn't describe it carefully enough, but it does work. This error also occurred in the demo above, as shown below, and was feared to cause unexpected problems! [demo](https://stackblitz.com/edit/vitejs-vite-rvdrxe?file=src%2FApp.jsx&terminal=dev)
![image](https://user-images.githubusercontent.com/53500371/161368487-81240ee6-1b3c-4b67-a3e3-dc9bcc979f67.png)

**Comment by layershifter** [comment]
I see, thanks for providing details 👍

It's not a problem with Vite, it's an issue with React itself:
https://codesandbox.io/s/admiring-darkness-fko8un?file=/src/App.js

It looks that validation below is not safe anymore:
https://github.com/microsoft/griffel/blob/05635e139c200e609ddb531f40e144b65f9cdcf3/packages/react/src/makeStyles.ts#L8-L16

---

I will look into it on next week. This is not okay.

**Comment by layershifter** [comment]
It looks that we can keep the check and do not produce errors 🎉 

```ts
function isInsideComponent() {
  // React 18 always logs errors if a dispatcher is not present,
  // so a check with hooks with always produce errors
  // https://github.com/facebook/react/blob/42f15b324f50d0fd98322c21646ac3013e30344a/packages/react/src/ReactHooks.js#L26-L36
  if (React["useInsertion" + "Effect"]) {
    return !!React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
      .ReactCurrentDispatcher.current;
  }

  try {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    React.useContext({});
    // React.useEffect();
    return true;
  } catch (e) {
    return false;
  }
}
```

P.S. I don't like to read React internals, but as it's a dev check it should be fine.

---

https://codesandbox.io/s/floral-bash-prwphq?file=/src/App.js

**Comment by itmanyong** [comment]
Thanks for the answer!

**Comment by thure** [comment]
I'm able to reproduce the issue as well, for what it's worth. I'm glad to see the issue is already identified.

**Comment by layershifter** [comment]
Was fixed in `@griffel/react@1.0.4`.
